tslearn/matrix_profile.py

```diff
@@ -130,21 +146,56 @@ def _transform(self, X, y=None):
         n_ts, sz, d = X.shape
         output_size = sz - self.subsequence_length + 1
         X_transformed = np.empty((n_ts, output_size, 1))
-        scaler = TimeSeriesScalerMeanVariance()
-        band_width = int(np.ceil(self.subsequence_length / 4))
-        for i_ts in range(n_ts):
-            segments = _series_to_segments(X[i_ts], self.subsequence_length)
-            if self.scale:
-                segments = scaler.fit_transform(segments)
-            n_segments = segments.shape[0]
-            segments_2d = segments.reshape((-1, self.subsequence_length * d))
-            dists = squareform(pdist(segments_2d, "euclidean"))
-            band = (np.tri(n_segments, n_segments,
-                           band_width, dtype=np.bool) &
-                    ~np.tri(n_segments, n_segments,
-                            -(band_width + 1), dtype=np.bool))
-            dists[band] = np.inf
-            X_transformed[i_ts] = dists.min(axis=1, keepdims=True)
+
+        if self.implementation == "stump":
+            import stumpy
+            if d > 1:
+                raise NotImplementedError("We currently don't support using "
+                                          "multi-dimensional matrix profiles "
+                                          "from the stumpy library.")
+            for i_ts in range(n_ts):
+                result = stumpy.stump(
+                    T_A=X[i_ts, :, 0].ravel(),
+                    m=self.subsequence_length)
+                X_transformed[i_ts, :, 0] = result[:, 0].astype(np.float)
+
+        elif self.implementation == "gpu_stump":
+            import stumpy
+            if d > 1:
+                raise NotImplementedError("We currently don't support using "
+                                          "multi-dimensional matrix profiles "
+                                          "from the stumpy library.")
+            for i_ts in range(n_ts):
+                result = stumpy.gpu_stump(
+                    T_A=X[i_ts, :, 0].ravel(),
+                    m=self.subsequence_length)
+                X_transformed[i_ts, :, 0] = result[:, 0].astype(np.float)
+
+        elif self.implementation == "numpy":
+            scaler = TimeSeriesScalerMeanVariance()
```

Comment:
Maybe we should check the dimension here too so that in all cases we rely on 1d data and we leave multivariate case as future works maybe (or do you plan to deal with it in this PR?)

Reply:
No, I don't plan on implementing the multivariate case in this PR.

Reply:
It probably makes sense to wait until users start asking for the multi-dimensional feature before adding it. I support your decision!